### PR TITLE
Adding certificates to the Pie object hierarchy

### PR DESCRIPTION
**Pull request by PeterHamilton**
This change adds certificate objects to the Pie object hierarchy, specifically a generic Certificate class and an X509Certificate subclass. A unit test suite covering both classes is included.

### CODE REVIEW

**Comment by PeterHamilton** [comment]
Forgot to add some stuff to this one, hold off on reviewing until I push another update.

**Comment by PeterHamilton** [comment]
Ok, this one should be ready to go now.

File: kmip/pie/objects.py [hunk 1]
```diff
@@ -599,3 +599,158 @@ def __ne__(self, other):
...
+class Certificate(CryptographicObject):
+    """
+    The Certificate class of the simplified KMIP object hierarchy.
+
+    A Certificate is a core KMIP object that is the subject of key management
+    operations. For more information, see Section 2.2 of the KMIP 1.1
+    specification.
+
+    Attributes:
+        certificate_type: The type of the Certificate.
```
Comment (rellerreller) [inline kmip/pie/objects.py]: Can you add class types to this?

File: kmip/pie/objects.py [hunk 2]
```diff
@@ -599,3 +599,158 @@ def __ne__(self, other):
             return not (self == other)
         else:
             return NotImplemented
+
+
+class Certificate(CryptographicObject):
```
Comment (rellerreller) [inline kmip/pie/objects.py]: Is there a way to mark this abstract or anything? We should add a note that this should not be invoked directly but use one of the subclasses.

**Comment by rellerreller** [comment]
A couple of small comments to address and then you should be good to go.

File: kmip/pie/objects.py (hunk 2 above)
Comment (PeterHamilton) [inline kmip/pie/objects.py]: Oh geeze, yeah, I should just use the `X509Certificate` here. I'll change this.

File: kmip/pie/objects.py (hunk 1 above)
Comment (PeterHamilton) [inline kmip/pie/objects.py]: What do you mean?

Comment (PeterHamilton) [inline kmip/pie/objects.py]: Oh, I guess you mean subclasses? If so, yes, you can now.

**Comment by PeterHamilton** [comment]
Ok, this has been updated. `Certificate` is now an uninstantiatable base class and all certificate operations should use `X509Certificate` instead.

File: kmip/pie/objects.py (hunk 1 above)
Comment (rellerreller) [inline kmip/pie/objects.py]: I meant in the description put certificate_type (class type here): The type...

Comment (rellerreller) [inline kmip/pie/objects.py]: Like you did on line 717.

**Comment by rellerreller** [comment]
One small comment about comments. After that is fixed then merge it.

File: kmip/pie/objects.py (hunk 1 above)
Comment (rellerreller) [inline kmip/pie/objects.py]: Never mind. I see what you did. I'll file this under id10t.

**Comment by rellerreller** [comment]
:+1: LGTM!

**Pull request merged**